mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -251,12 +252,79 @@ def _list_experiments(
         query_options = self._get_eager_experiment_query_options() if eager else []
         return session.query(SqlExperiment).options(*query_options).filter(*conditions).all()
 
-    def list_experiments(self, view_type=ViewType.ACTIVE_ONLY):
+    def _list_experiments(
+        self,
+        ids=None,
+        names=None,
+        view_type=ViewType.ACTIVE_ONLY,
+        max_results=None,
+        page_token=None,
+        eager=False,
+    ):
+        """
+        :param max_results: If passed, specifies the maximum number of experiments desired. If not
+                            passed, all experiments will be returned.
+        :param page_token: Token specifying the next page of results. It should be obtained from
+                            a ``list_experiments`` call.
+        :param eager: If ``True``, eagerly loads each experiments's tags. If ``False``, these tags
+                      are not eagerly loaded and will be loaded if/when their corresponding
+                      object properties are accessed from a resulting ``SqlExperiment`` object.
+        """
+        stages = LifecycleStage.view_type_to_stages(view_type)
```

Comment:
Could we add validation to make sure that `max_results` is below our limit (e.g. 10,000)? For reference see [the helper method for validating params](https://github.com/mlflow/mlflow/blob/ed2b8d7359e053c1b3ded54c3a2c0c064ccf347b/mlflow/utils/validation.py#L151) and how we use it (we basically apply it in each of our built-in store implementations)

Reply:
And you can add tests in [this file](https://github.com/mlflow/mlflow/blob/ed2b8d7359e053c1b3ded54c3a2c0c064ccf347b/tests/utils/test_validation.py#L65-L71)